syncapi/storage/sqlite3/peeks_table.go

```diff
@@ -27,13 +27,14 @@ import (
 
 const peeksSchema = `
 CREATE TABLE IF NOT EXISTS syncapi_peeks (
-	id INTEGER PRIMARY KEY,
+	id INTEGER,
 	room_id TEXT NOT NULL,
 	user_id TEXT NOT NULL,
 	device_id TEXT NOT NULL,
 	deleted BOOL NOT NULL DEFAULT false,
     -- When the peek was created in UNIX epoch ms.
-    creation_ts INTEGER NOT NULL
+    creation_ts INTEGER NOT NULL,
+    UNIQUE(room_id, user_id, device_id)
```

Comment:
This alone will still cause problems if you peek/unpeek/peek into the same room I think. The 2nd peek will fail to insert.

Reply:
fixed by https://github.com/matrix-org/dendrite/pull/1370/commits/956184377276dd91e82e0103ffe24c9c8ddc0fd2